Client/Ecosia/UI/MultiplyImpact/MultiplyImpact.swift

```diff
@@ -508,7 +508,7 @@ final class MultiplyImpact: UIViewController, NotificationThemeable {
     private func updateInviteLink() {
         guard let inviteLink = inviteLink else {
             copyLink?.text = "-"
-            referrals.refresh(createCode: true) {[weak self] error in
+            referrals.refresh(force: true, createCode: true) {[weak self] error in
```

Comment:
At the moment, I see the only thing we do after the refresh is call `self?.updateInviteLink()` again so that `copyLink?.text = inviteLink` can be set. Since we are now refreshing the referrals and potentially get new claims, we should also refresh the counter when there is an update. I would even suggest properly subscribing to `referrals` since it is a Publisher and does already have a logic of sending whenever the count updates.

Also this might be a bit conflicting with the changes made on https://github.com/ecosia/ios-browser/pull/568 since there a new component for the referral counter is used that would need to be updated 🤔 I guess it all depends on which is merged first to see how it is handled.

Reply:
As I'll be planning to move the refresh logic out of the `updateInviteLink()` we can also double-check whether this is still needed.